### PR TITLE
improved content retrieval by adjusting _clean_data function

### PR DESCRIPTION
**Pull request by gisturiz**

### CODE REVIEW

File: libs/langchain/langchain/document_loaders/readthedocs.py [hunk 1]
```diff
@@ -3,11 +3,13 @@
 
 from langchain.docstore.document import Document
 from langchain.document_loaders.base import BaseLoader
+import re
 
 
-class ReadTheDocsLoader(BaseLoader):
+class ReadDocLoader(BaseLoader):
```
Comment (eyurtsev) [inline libs/langchain/langchain/document_loaders/readthedocs.py:9]: Renaming the loader is a breaking change (will break existing users code)

File: libs/langchain/langchain/document_loaders/readthedocs.py [hunk 2]
```diff
@@ -67,33 +70,14 @@ def load(self) -> List[Document]:
...
-                break
-
-        if text is not None:
-            text = text.get_text()
-        else:
-            text = ""
-        # trim empty lines
-        return "\n".join([t for t in text.split("\n") if t])
+           
+        text = soup.get_text().strip()
```
Comment (eyurtsev) [inline libs/langchain/langchain/document_loaders/readthedocs.py:79]: It's not clear why this code would result in better results. In order to get these changes accepted, you'd need to add unit tests showing different scenarios where the parse results in better data.

**Review by eyurtsev — CHANGES_REQUESTED** [review]
Thank you for the contribuion! This PR invovles breaking changes (name of loader) which we definitely don't want to do. It also contains changes to cleaning logic, in order to accept that there needs to be some proof that the logic is actually better than existing logic.

File: libs/langchain/langchain/document_loaders/readthedocs.py (hunk 1 above)
Comment (gisturiz) [reply libs/langchain/langchain/document_loaders/readthedocs.py:9]: Apologies for this oversight, I was testing the original vs my implementation which is why I had changed the name. This is taken care of now, working on the unit tests

**Pull request closed without merging**